### ISSUE TITLE
ignore.test failure on armel for 1.3.4

### ISSUE THREAD

**Issue by reneengelhard** [hunspell/hunspell#370]
Hi,

see https://buildd.debian.org/status/fetch.php?pkg=hunspell&arch=armel&ver=1.3.4-1&stamp=1460072167:

```
[...]
PASS: affixes.test
PASS: condition.test
PASS: condition_utf.test
PASS: base.test
PASS: base_utf.test
PASS: allcaps.test
PASS: allcaps_utf.test
PASS: allcaps2.test
PASS: allcaps3.test
PASS: keepcase.test
PASS: i58202.test
PASS: map.test
PASS: rep.test
PASS: sug.test
PASS: sugutf.test
PASS: phone.test
PASS: flag.test
PASS: flaglong.test
PASS: flagnum.test
PASS: flagutf8.test
PASS: slash.test
PASS: forbiddenword.test
PASS: nosuggest.test
PASS: alias.test
PASS: alias2.test
PASS: alias3.test
PASS: breakdefault.test
PASS: break.test
PASS: needaffix.test
PASS: needaffix2.test
PASS: needaffix3.test
PASS: needaffix4.test
PASS: needaffix5.test
PASS: circumfix.test
PASS: fogemorpheme.test
PASS: onlyincompound.test
PASS: complexprefixes.test
PASS: complexprefixes2.test
PASS: complexprefixesutf.test
PASS: conditionalprefix.test
PASS: zeroaffix.test
PASS: utf8.test
PASS: utf8_bom.test
PASS: utf8_bom2.test
PASS: utf8_nonbmp.test
PASS: compoundflag.test
PASS: compoundrule.test
PASS: compoundrule2.test
PASS: compoundrule3.test
PASS: compoundrule4.test
PASS: compoundrule5.test
PASS: compoundrule6.test
PASS: compoundrule7.test
PASS: compoundrule8.test
PASS: compoundaffix.test
PASS: compoundaffix2.test
PASS: compoundaffix3.test
PASS: checkcompounddup.test
PASS: checkcompoundtriple.test
PASS: simplifiedtriple.test
PASS: checkcompoundrep.test
PASS: checkcompoundcase2.test
PASS: checkcompoundcaseutf.test
PASS: checkcompoundpattern.test
PASS: checkcompoundpattern2.test
PASS: checkcompoundpattern3.test
PASS: checkcompoundpattern4.test
PASS: utfcompound.test
PASS: checksharps.test
PASS: checksharpsutf.test
PASS: germancompounding.test
PASS: germancompoundingold.test
PASS: i35725.test
PASS: i53643.test
PASS: i54633.test
PASS: i54980.test
PASS: maputf.test
PASS: reputf.test
FAIL: ignore.test
PASS: ignoreutf.test
PASS: 1592880.test
PASS: 1695964.test
PASS: 1463589.test
PASS: 1463589_utf.test
PASS: IJ.test
PASS: i68568.test
PASS: i68568utf.test
PASS: 1706659.test
PASS: digits_in_words.test
PASS: colons_in_words.test
PASS: ngram_utf_fix.test
PASS: morph.test
PASS: 1975530.test
PASS: fullstrip.test
PASS: iconv.test
PASS: oconv.test
PASS: encoding.test
PASS: korean.test
PASS: opentaal_forbiddenword1.test
PASS: opentaal_forbiddenword2.test
PASS: opentaal_keepcase.test
PASS: arabic.test
PASS: 2970240.test
PASS: 2970242.test
PASS: breakoff.test
PASS: opentaal_cpdpat.test
PASS: opentaal_cpdpat2.test
PASS: 2999225.test
PASS: onlyincompound2.test
PASS: forceucase.test
PASS: warn.test
============================================================================
Testsuite summary for hunspell 1.3.4
============================================================================
# TOTAL: 111
# PASS:  110
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See tests/test-suite.log
Please report to nemeth@numbertext.org
============================================================================
```

The other archs built fine, see https://buildd.debian.org/status/logs.php?pkg=hunspell&ver=1.3.4-1&suite=sid

Log is:

```
# cat tests/test-suite.log
==========================================
   hunspell 1.3.4: tests/test-suite.log
==========================================

# TOTAL: 111
# PASS:  110
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: ignore
============

=============================================
Fail in ignore.good. Good words recognised as wrong:
example
expression
xmpl
xprssn
reexpression
rxprssn
FAIL ignore.test (exit status: 1)
```

**Comment by reneengelhard** [comment]
@caolanm FTR, this works with master

**Comment by caolanm** [comment]
This may have been fixed with 2824e4ed8fe9c476234edba87bf6042ca862ec8f "missing the last argument to std::string::erase"

**Comment by reneengelhard** [comment]
@caolanm indeed, thanks
